pkg/resource/deploy/step_generator.go

```diff
@@ -1976,28 +1977,34 @@ func (sg *stepGenerator) GenerateDeletes(targetsOpt UrnTargets, excludesOpt UrnT
 					logging.V(7).Infof("Planner decided to delete '%v' due to replacement", res.URN)
 					sg.deletes[res.URN] = true
 					oldViews := sg.deployment.GetOldViews(res.URN)
-					steps = append(steps, NewDeleteReplacementStep(sg.deployment, sg.deletes, res, false, oldViews))
+					deleteSteps = append(deleteSteps, NewDeleteReplacementStep(sg.deployment, sg.deletes, res, false, oldViews))
 				} else if !sg.isOperatedOn(res.URN) {
 					logging.V(7).Infof("Planner decided to delete '%v'", res.URN)
 					sg.deletes[res.URN] = true
 					if !res.PendingReplacement {
 						oldViews := sg.deployment.GetOldViews(res.URN)
-						steps = append(steps, NewDeleteStep(sg.deployment, sg.deletes, res, oldViews))
+						deleteSteps = append(deleteSteps, NewDeleteStep(sg.deployment, sg.deletes, res, oldViews))
 					} else {
-						steps = append(steps, NewRemovePendingReplaceStep(sg.deployment, res))
+						deleteSteps = append(deleteSteps, NewRemovePendingReplaceStep(sg.deployment, res))
 					}
 				}
 
 				// We just added a Delete step, so we need to ensure the provider for this resource is available.
 				if sg.deletes[res.URN] {
 					err := sg.deployment.EnsureProvider(res.Provider)
 					if err != nil {
-						return nil, fmt.Errorf("could not load provider for resource %v: %w", res.URN, err)
+						return nil, nil, fmt.Errorf("could not load provider for resource %v: %w", res.URN, err)
 					}
 				}
 			} else {
-				// Add this resource to the deployment.news so that it shows up in analysis.
-				sg.deployment.news.Store(res.URN, res)
+				// If we get here this is a resource in the state that we're skipping over due to targets. If
+				// we had registered it from the program it would have become a SameStep, so lets replicate
+				// that here.
+				if !sg.isOperatedOn(res.URN) {
+					new := res.Copy()
+					new.ID = ""
```

Comment:
Why do we reset the ID here? I guess that's only relevant for providers, but wouldn't this break providers that are same stepped here?

Reply:
SameSteps assert that new.ID is "" but then just copy old.ID over to it. Given that new here is just a copy of old that all that results in just the same thing.